hummingbird/ml/_container.py

```diff
@@ -319,12 +330,65 @@ def score_samples(self, *inputs):
 
 
 # PyTorch containers.
-class PyTorchSklearnContainer(ABC):
+class PyTorchSklearnContainer(SklearnContainer):
     """
     Base container for PyTorch models.
-    We used this container to surface PyTorch-specific functionalities in the containers.
     """
 
+    def save(self, location):
+        assert self.model is not None, "Saving a None model is undefined."
+
+        if constants.TEST_INPUT in self._extra_config:
+            self._extra_config[constants.TEST_INPUT] = None
+
+        if "torch.jit" in str(type(self.model)):
+            assert not os.path.exists(location), "Directory {} already exists.".format(location)
+            os.makedirs(location)
+            self.model.save(os.path.join(location, "deploy_model.zip"))
+            model = self.model
+            self._model = None
+            with open(os.path.join(location, "container.pkl"), "wb") as file:
+                dill.dump(self, file)
+            self._model = model
+        else:
+            if not location.endswith("pkl"):
+                location += "pkl"
+            assert not os.path.exists(location), "File {} already exists.".format(location)
+            with open(location, "wb") as file:
+                dill.dump(self, file)
+
+    @staticmethod
+    def load(location):
+        """
+        Method used to load a container from the file system.
+
+        Args:
+            location: The location on the file system where to load the model.
+
+        Returns:
+            The loaded model.
+        """
+        assert os.path.exists(location), "Model location {} does not exist.".format(location)
+
+        container = None
+        if os.path.isdir(location):
+            # This is a torch.jit model
+            model = torch.jit.load(os.path.join(location, "deploy_model.zip"))
+            with open(os.path.join(location, "container.pkl"), "rb") as file:
+                container = dill.load(file)
+            container._model = model
+        else:
+            # This is a pytorch  model
+            with open(location, "rb") as file:
+                container = dill.load(file)
+
+        if container._n_threads is not None:
+            if torch.get_num_interop_threads() != 1:
```

Comment:
Can you please put a comment here explaining why? :)

Reply:
Fixed.